.github/workflows/pull_request.yaml

```diff
@@ -0,0 +1,51 @@
+name: Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  jdk17:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup Java
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Cache local Maven repository
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: Checkout Infinispan repo
+        uses: actions/checkout@v2
+        with:
+          repository: infinispan/infinispan
+          path: infinispan
+          ref: ${{ github.event.pull_request.base.ref }}
+
+      - name: Build Infinispan SNAPSHOT
```

Comment:
why do we need the snapshot?

Reply:
Because `version.infinispan` is `15.0.0-SNAPSHOT` and we don't publish SNAPSHOTS, maybe we should?

Reply:
I would say to change it to 15.0.0.Dev01. Except if you are going to trigger this GH action for each commit in the Infinispan repo, it does not worth the extra work to support SNAPSHOT.
Also, I don't think it is worth releasing this repo if there is no code change in it since `infinispan-core` is marked as provided. 
Just my 2 cents :dollar:

Reply:
> I would say to change it to 15.0.0.Dev01. Except if you are going to trigger this GH action for each commit in the Infinispan repo, it does not worth the extra work to support SNAPSHOT.

+1, I think that's a better option.

> Also, I don't think it is worth releasing this repo if there is no code change in it since `infinispan-core` is marked as provided.

I think that's fine once we have a `.Final` release, i.e. a 15.0.0.Final release with `version.infinispan=15.0.0.Final`, as SPI incompatibility shouldn't be broken outside of a major version increment.

Reply:
Updated to use a released Infinispan version.